network.go

```diff
@@ -1643,9 +1653,26 @@ func (n *network) ResolveName(req string, ipType int) ([]net.IP, bool) {
 	return nil, ipv6Miss
 }
 
-func (n *network) ResolveIP(ip string) string {
-	var svc string
+func (n *network) HandleQueryResp(name string, ip net.IP) {
+	c := n.getController()
+	c.Lock()
+	sr, ok := c.svcRecords[n.ID()]
+	c.Unlock()
+
+	if !ok {
+		return
+	}
+
+	ipStr := netutils.ReverseIP(ip.String())
+
+	n.Lock()
```

Comment:
sr which is of type `svcInfo` is a per network structure. In the lookup function I am taking a network level lock to avoid concurrent read & write to the maps that are members of `svcInfo`.  But looking at the code we are not taking lock in the functions `addNameToIP`, `addIPToName` etc.  I think this needs to be fixed.